Tensile/KernelWriter.py

```diff
@@ -848,11 +858,166 @@ def kernelBody( self, kernel, tensorParametersA, tensorParametersB ):
+                 elif iter+1 != numGroups:
+                   waitCntItems[iter] = TotalnumLdsFetches - ((ldsItems[iter][0])/kernel["VectorWidth"] + (ldsItems[iter][1])/kernel["VectorWidth"] + waitCntItems[iter-1])
+                 else:
+                   waitCntItems[iter] = 0
+                 #print("Waitcnt(%u..%u)\n"%(iter,waitCntItems[iter]))
+             for iter in range(0,numGroups):
+               #Mac Code
+               #place holder for future work Instruction class for generting MAC instruction
+               #FMAInstruction = MacInstruction(globalParameters["CurrentISA"])
+               subIterCode = Code.Module()
+               waitCode = Code.Module()
+               macIterCodeGrp = Code.Module()
+               doOnce = False
+               if self.enable["MAC"]:
+                 numMacItems = macItems[iter]
+                 for Items in range(0,numMacItems):
+                   macItem = MacitemsReorder.pop(0)
+                   macIterCodeGrp.addCode(macItem)
+                   ## add s_setprio 1 when AggressivePerfMode ==1 as second instruction for second-last blob macCode
+                   if (kernel["AggressivePerfMode"] and not doOnce):
+                       macIterCodeGrp.addInst("s_setprio ","1","Raise priority while processing macs")
+                       doOnce = True
+                 ## add s_setprio 0 when AggressivePerfMode ==1 as last instruction
+                 if (kernel["AggressivePerfMode"]):
+                   macIterCodeGrp.addInst("s_setprio ","0","Reset priority after macs")
+               #print("ReadWaitcnt(%u..%u)\n"%(iter,waitCntItems[iter]))
+               #print("WriteCodeCount(%d..%u)\n",u,self.perIterLocalWriteCode[u].count())
+               if (iter == 0):
+                 if self.enable["Wait"]:
+                   #calculate lgkmcnt value including read+write for first iteration
+                   waitCntVal = waitCntItems[iter] + 1 if (self.perIterLocalWriteCode[u].count()>0) else waitCntItems[iter]
+                   # read + write instructions lgkmcnt (1=> for write)
+                   # build waitCnt using new lgkmcnt
+                   waitCode = Code.WaitCnt(waitCntVal,-1,"wait for prior local read")
+                 subIterCode = self.makeSubIterSchedule(kernel, localReads, \
+                          self.perIterGlobalReadCode[u], self.perIterLocalWriteCode[u],
+                          pointerCode, waitCode, macIterCodeGrp)
+               else:
+                   #last group only pointer + localWrite Code
+                 if self.enable["Wait"]:
+                   waitCode = Code.WaitCnt(waitCntItems[iter],-1,"wait for prior local read & local writes")
+                 subIterCode.addCode(waitCode)
+                 subIterCode.addCode(macIterCodeGrp)
+               kl.append(subIterCode) # add scheduled "other", local reads, local writes
```

Comment:
this only runs for Asm path correct?

Reply:
seems this scheduling code should exist in a separate function (to isolate the code and improve maintainability)